src/context/AuthContext/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { openInfinityWallet } from '@infinitywallet/infinity-connector';
+import type { Provider } from '@wagmi/core';
+import config from 'config';
+import { Signer, getDefaultProvider } from 'ethers';
+import noop from 'noop-ts';
+import React, { useCallback, useContext, useState } from 'react';
+import { useTranslation } from 'translation';
+import { useAccount, useConnect, useDisconnect, useProvider, useSigner } from 'wagmi';
+
+import { Connector, connectorIdByName } from 'clients/web3';
+import { AuthModal } from 'components/AuthModal';
+import { LS_KEY_CONNECTED_CONNECTOR } from 'constants/localStorageKeys';
+import getConnectedConnector from 'context/AuthContext/getConnectedConnector';
+import useCopyToClipboard from 'hooks/useCopyToClipboard';
+import { isRunningInInfinityWalletApp } from 'utilities/walletDetection';
+
+export interface Account {
+  address: string;
+  connector?: Connector;
+}
+
+export interface AuthContextValue {
+  login: (connector: Connector) => Promise<void>;
+  logOut: () => void;
+  openAuthModal: () => void;
+  closeAuthModal: () => void;
+  provider: Provider;
+  account?: Account;
+  signer?: Signer;
+}
+
+export const AuthContext = React.createContext<AuthContextValue>({
+  login: noop,
+  logOut: noop,
+  openAuthModal: noop,
+  closeAuthModal: noop,
+  provider: getDefaultProvider(),
+});
+
+export const AuthProvider: React.FC = ({ children }) => {
+  const [isAuthModalOpen, setIsAuthModalOpen] = React.useState(false);
+  const { connectors, connectAsync } = useConnect();
+  const { disconnectAsync } = useDisconnect();
+  const { data: signer } = useSigner();
+  const provider = useProvider();
+  const { address, status } = useAccount();
+
+  const [connectedConnector, setConnectedConnector] = useState(getConnectedConnector());
+
+  const login = useCallback(async (connectorId: Connector) => {
+    // If user is attempting to connect their Infinity wallet but the dApp
+    // isn't currently running in the Infinity Wallet app, open it
+    if (connectorId === Connector.InfinityWallet && !isRunningInInfinityWalletApp()) {
+      openInfinityWallet(window.location.href, config.chainId);
+      return;
+    }
+
+    // TODO: determine based on connector ID passed
```

Comment:
Do we need to address this TODO?

Reply:
No, it should have been removed (the logic to do that is on the next line).

Reply:
ahh I see I misunderstood I thought this was an extra property that multicall needed. You can still solve this by upgrading ethers to match the version (5.7.2) used by multicall